components/pokemonapp.brs

```diff
@@ -32,19 +37,40 @@ function observePokeList()
     m.pokeList.observeField("rowItemSelected", "onPokemonSelect")
 end function
 
+function observeRatingList()
+    m.pokeRatingList.observeField("rowItemSelected", "onRatingSelect")
+end function
+
+function onRatingSelect(event as Object)
+    data = event.getData()
+    ratingScore = data[m.focusedItemIndex] + 1
+    currentRow = m.pokeList.content.getChild(m.firstChild)
+    selectedPokemon = currentRow.getChild(m.focusedPokemonIndex)
+    selectedPokemon.pokemonRating = ratingScore
+    m.registry.Write(selectedPokemon.pokemonID.ToStr(),ratingScore.ToStr())
+    m.registry.Flush()
```

Comment:
This part can be done separately in order to keep the code clean and make the functionality here more easy to understand.
https://refactoring.guru/smells/long-method